Could not resolve "../package.json" on v5.20.1

Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

1) Is this a client library issue or a product issue?
library issue

2) Did someone already solve this?
Not that I can find

3) Do you have a support contract?
No

#### Environment details

  - OS: macOS
  - Node.js version: 14 and 16
  - npm version: 6.14.16 and 8.5.5
  - `@google-cloud/storage` version: v5.20.1

#### Steps to reproduce

  1. Install `@google-cloud/storage` and use it as a dependency in your project
  2. Package your project with esbuild
  3. See that it fails with the following errors:
```
✘ [ERROR] Could not resolve "../package.json"

    node_modules/@google-cloud/storage/build/src/storage.js:384:34:
      384 │             packageJson = require('../package.json');
          ╵                                   ~~~~~~~~~~~~~~~~~

✘ [ERROR] Could not resolve "../../package.json"

    node_modules/@google-cloud/storage/build/src/nodejs-common/util.js:37:26:
      37 │     packageJson = require('../../package.json');
         ╵                           ~~~~~~~~~~~~~~~~~~~~

✘ [ERROR] Could not resolve "../package.json"

    node_modules/@google-cloud/storage/build/src/gcs-resumable-upload.js:37:26:
      37 │     packageJson = require('../package.json');
         ╵                           ~~~~~~~~~~~~~~~~~
```

I am using `firebase-admin` which has `@google-cloud/storage` as a dependency. When I pin `@google-cloud/storage` to `5.19.4` I can package my app fine with esbuild. When it is updated to `v5.20.1` it starts to fail with the above error.

--- a/src/gcs-resumable-upload.ts
+++ b/src/gcs-resumable-upload.ts
@@ -28,6 +28,7 @@ import {Readable, Writable} from 'stream';
 import retry = require('async-retry');
 import {RetryOptions, PreconditionOptions} from './storage';
 import * as uuid from 'uuid';
+import path = require('path');
 
 const NOT_FOUND_STATUS_CODE = 404;
 const TERMINATED_UPLOAD_STATUS_CODE = 410;
@@ -36,10 +37,10 @@ const DEFAULT_API_ENDPOINT_REGEX = /.*\.googleapis\.com/;
 let packageJson: ReturnType<JSON['parse']> = {};
 try {
   // if requiring from 'build' (default)
-  packageJson = require('../../package.json');
+  packageJson = require(path.join(__dirname, '../../package.json'));
 } catch (e) {
   // if requiring directly from TypeScript context
-  packageJson = require('../package.json');
+  packageJson = require(path.join(__dirname, '../package.json'));
 }
 
 export const PROTOCOL_REGEX = /^(\w*):\/\//;

--- a/src/nodejs-common/util.ts
+++ b/src/nodejs-common/util.ts
@@ -29,13 +29,14 @@ import {Duplex, DuplexOptions, Readable, Transform, Writable} from 'stream';
 import {teenyRequest} from 'teeny-request';
 import {Interceptor} from './service-object';
 import * as uuid from 'uuid';
+import path = require('path');
 let packageJson: ReturnType<JSON['parse']> = {};
 try {
   // if requiring from 'build' (default)
-  packageJson = require('../../../package.json');
+  packageJson = require(path.join(__dirname, '../../../package.json'));
 } catch (e) {
   // if requiring directly from TypeScript context
-  packageJson = require('../../package.json');
+  packageJson = require(path.join(__dirname, '../../package.json'));
 }
 
 // eslint-disable-next-line @typescript-eslint/no-var-requires

--- a/src/storage.ts
+++ b/src/storage.ts
@@ -23,6 +23,7 @@ import {Channel} from './channel';
 import {File} from './file';
 import {normalize} from './util';
 import {HmacKey, HmacKeyMetadata, HmacKeyOptions} from './hmacKey';
+import path = require('path');
 
 export interface GetServiceAccountOptions {
   userProject?: string;
@@ -618,10 +619,10 @@ export class Storage extends Service {
 
     try {
       // if requiring from 'build' (default)
-      packageJson = require('../../package.json');
+      packageJson = require(path.join(__dirname, '../../package.json'));
     } catch (e) {
       // if requiring directly from TypeScript context
-      packageJson = require('../package.json');
+      packageJson = require(path.join(__dirname, '../package.json'));
     }
 
     const config = {

